### ISSUE TITLE
A11y: Role is incorrect for single select Details List

### ISSUE THREAD

**Issue by drotara** [microsoft/fluentui#16681]
### Environment Information

- **Package version(s)**: 7.153.2 +
- **Browser and OS versions**: Edge 88.0.705.50 

### Please provide a reproduction of the issue in a codepen:
NOTE: None of the "Export to Codepen" for details list works -_-... 
Here is a VERY basic example of the issue: https://codepen.io/adrotar/pen/YzpKZjg?editors=0010

#### Actual behavior:

The role on the Radio button is "checkbox"
![image](https://user-images.githubusercontent.com/2686972/106073018-283bea00-60be-11eb-8763-5210d5c7ae2c.png)

#### Expected behavior:
The role when in single select mode should be "Radio" 

### Documentation describing expected behavior
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc=%7b248054a6-5e68-4771-9e1e-242fb5025730%7d

**Comment by gouttierre** [comment]
@drotara - Thanks for the information and share. I see the repro as you stated.

@khmakoto - Would you be able to confirm if this is a regression, or if this behavior is an issue where role should be Radio?
cc @smhigley
